src/Stratis.Bitcoin/Consensus/Rules/ConsensusRules.cs

```diff
@@ -168,6 +185,53 @@ public void SetErrorHandler(ConsensusRule errorHandler)
             }
         }
 
+        /// <inheritdoc/>
+        public async Task FullValidationAsync(ValidationContext validationContext, ChainedHeader tip)
+        {
+            Guard.NotNull(validationContext, nameof(validationContext));
+
+            RuleContext ruleContext = this.CreateRuleContext(validationContext, tip);
+
+            await this.ExecuteRulesAsync(this.partialValidationRules, ruleContext);
+        }
+
+        /// <inheritdoc/>
+        public async Task PartialValidationAsync(ValidationContext validationContext, ChainedHeader tip)
+        {
+            Guard.NotNull(validationContext, nameof(validationContext));
+
+            RuleContext ruleContext = this.CreateRuleContext(validationContext, tip);
+
+            await this.ExecuteRulesAsync(this.partialValidationRules, ruleContext);
+        }
+
+        private async Task ExecuteRulesAsync(List<ConsensusRuleDescriptor> rules, RuleContext ruleContext)
+        {
+            try
+            {
+                using (new StopwatchDisposable(o => this.PerformanceCounter.AddBlockProcessingTime(o)))
+                {
+                    ruleContext.SkipValidation = ruleContext.ValidationContext.ChainedHeader.BlockValidationState == ValidationState.AssumedValid;
+
+                    foreach (ConsensusRuleDescriptor ruleDescriptor in rules)
+                    {
+                        if (ruleContext.SkipValidation && ruleDescriptor.CanSkipValidation)
+                        {
+                            this.logger.LogTrace("Rule {0} skipped for block at height {1}.", nameof(ruleDescriptor), ruleContext.ConsensusTip?.Height);
```

Comment:
Can the consensus to be null?

Reply:
I think it can as its set in one of the rules but this is temporary as when we upgrade to CM this will not be null (unless its a trx validation but not sure how this will look yet).